iree/turbine/kernel/wave/layers/quant_attention.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2025 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import torch
+import math
+import functools
+import iree.turbine.kernel.lang as tkl
+from iree.turbine.kernel.wave.utils.general_utils import (
+    get_default_scheduling_params,
+)
+from iree.turbine.kernel.wave.utils.run_utils import (
+    set_default_run_config,
+)
+from iree.turbine.kernel.wave.templates.quantized_attention import (
+    get_brevitas_pertensor_fp8_attention_kernel,
+)
+from iree.turbine.kernel.wave.scheduling.schedule import SchedulingType
+from iree.turbine.kernel.wave.compile import WaveCompileOptions, wave_compile
+from iree.turbine.kernel.wave.constraints import MMAType
+from iree.turbine.kernel.wave.templates.attention_common import AttentionShape
+
+
+ATTENTION_SUPPORTED_DTYPE = {torch.float16}
+
+
+@functools.lru_cache
+def get_wave_kernel(
+    shape: AttentionShape,
+    q_scale: float,
+    k_scale: float,
+    v_scale: float,
+    logit_dtype: torch.dtype,
+    is_causal: bool,
+):
+    assert shape.num_query_heads % shape.num_kv_heads == 0
+
+    mfma_variant = (MMAType.F32_16x16x32_F8, MMAType.F32_16x16x32_K4_F8)
+    (
+        quantized_attention,
+        hyperparams,
+        _,
+        _,
+    ) = get_brevitas_pertensor_fp8_attention_kernel(
+        shape,
+        mfma_variant,
+        q_scale=q_scale,
+        k_scale=k_scale,
+        v_scale=v_scale,
+        is_causal=is_causal,
+        logit_dtype=logit_dtype,
+    )
+    hyperparams.update(get_default_scheduling_params())
+    del hyperparams[tkl.sym.B]
+    del hyperparams[tkl.sym.N_Q]
+    del hyperparams[tkl.sym.N_KV]
+    dynamic_symbols = [tkl.sym.B, tkl.sym.N_Q, tkl.sym.N_KV]
+
+    options = WaveCompileOptions(
+        subs=hyperparams,
+        schedule=SchedulingType.NONE,
+        use_scheduling_barriers=False,
+        dynamic_symbols=dynamic_symbols,
+        dynamic_symbols_map={},
+        waves_per_eu=2,
+        denorm_fp_math_f32="preserve-sign",
+    )
+    options = set_default_run_config(options)
+    quantized_attention = wave_compile(options, quantized_attention)
+    return quantized_attention
+
+
+def wave_sdpa_fp8(query, key, value, q_scale, k_scale, v_scale, is_causal=False):
```

Comment:
I think it would be good to have the dtype (`fp8`) as a parameter and have more general "quant attention" layer to expand to other quant types down the line.

Reply:
What do you think of wrapping this function as a `nn.Module` like https://github.com/iree-org/iree-turbine/blob/main/iree/turbine/kernel/wave/layers/quant_linear.py#L199 ?

Reply:
> I think it would be good to have the dtype (fp8) as a parameter and have more general "quant attention" layer to expand to other quant types down the line.

We can do this in the following PR

>What do you think of wrapping this function as a nn.Module like https://github.com/iree-org/iree-turbine/blob/main/iree/turbine/kernel/wave/layers/quant_linear.py#L199 ?

Per offline discussion and the landed PR, we decided to move this to nn.functional